Mathlib/CategoryTheory/Limits/Fubini.lean

```diff
@@ -195,6 +195,50 @@ def coneOfConeUncurryIsLimit {D : DiagramOfCones F} (Q : ∀ j, IsLimit (D.obj j
     rw [← w j]
     simp
 
+/-- Conversely, if `coneOfConeUncurry Q c` is a limit cone then `c` is in fact a limit cone.
+-/
+def isLimitOfConeOfConeUncurryIsLimit {D : DiagramOfCones F} (Q : ∀ j, IsLimit (D.obj j))
+    {c : Cone (uncurry.obj F)} (P : IsLimit (coneOfConeUncurry Q c)) : IsLimit c :=
+  -- These constructions are used in various fields of the proof so we abstract them here.
+  letI E : (j : J) → (Prod.sectR j K ⋙ uncurry.obj F ≅ F.obj j) := fun j =>
+    NatIso.ofComponents (fun _ ↦ Iso.refl _)
```

Comment:
Untested, but we now have the following convenient syntax:
```suggestion
  letI E (j : J) : Prod.sectR j K ⋙ uncurry.obj F ≅ F.obj j :=
    NatIso.ofComponents (fun _ ↦ Iso.refl _)
```

Reply:
I like this new syntax!